airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -178,7 +178,7 @@ def _request_headers(
             stream_slice,
             next_page_token,
             self._paginator.get_request_headers,
-            self.stream_slicer.get_request_headers,
+            self.request_option_provider.get_request_headers,
```

Comment:
@brianjlai was there a reason we were still using the `stream_slicer` here instead of the `request_option_provider`?